R/teal_slice.R

```diff
@@ -1,301 +1,207 @@
-#' Manage filter state(s).
+#' Filter specification
 #'
-#' Functions for passing filter state information between objects.
+#' @details
+#' `teal_slice` object fully describes filter state and can be used to create,
+#' modify, and delete a filter state. A `teal_slice` contains a number of common fields
+#' (all named arguments of `teal_slice`), some of which are mandatory, but only
+#' `dataname` and  either `varname` or `expr` must be specified, while the others have default
+#' values.
 #'
-#' These functions create and manage filter state specifications.
-#' A single filter state can be fully described by a `teal_slice` object and such
-#' objects will be used to create, modify, and delete a filter state.
-#'
-#' A `teal_slice` contains a number of common fields (all named arguments of `filter_var`),
-#' some of which are mandatory, but only `dataname` and `varname` must be specified,
-#' while the others have default values.
 #' Setting any of the other values to NULL means that those properties will not be modified
 #' (when setting an existing state) or that they will be determined by data (when creating new a new one).
-#' Each of the common fields corresponds to one private field in `FilterState`
-#' where it is stored and from where it is retrieved when calling `FiterState$get_state`.
-#'
-#' A `teal_slice` can also contain any number of additional fields, passed to `...`
-#' as `name:value` pairs. These are collated into a list and stored in the
-#' `private$extras` field in `FilterState`.
+#' Entire object is `FilterState` class member and can be accessed with `FilterState$get_state()`.
+#'
+#' A `teal_slice` can come in two flavors:
+#' 1. `teal_slice_var` -
+#' this describes a typical interactive filter that refers to a single variable, managed by the `FilterState` class.
+#' This class is created when `varname is specified.
+#' The object retains all fields specified in the call. `id` can be created by default and need not be specified.
+#' 2. `teal_slice_expr` -
+#' this describes a filter state that refers to an expression, which can potentially include multiple variables,
+#' managed by the `FilterStateExpr` class.
+#' This class is created when `expr` is specified.
+#' `dataname` and `locked` are retained, `fixed` is set to `TRUE`, `id` becomes mandatory, `title`
+#' remains optional, while other arguments are disregarded.
 #'
 #' All `teal_slice` fields can be passed as arguments to `FilterState` constructors.
```

Comment:
```suggestion
```
This is not true anymore, the whole `teal_slices` is passed, not individual fields.